opacus/privacy_engine/factory.py

```diff
@@ -0,0 +1,33 @@
+from opacus import PrivacyEngine
+from opacus.privacy_engine.catalog import (
+    PrivacyEngine,
+    PrivacyEngineNonPoisson,
+    PrivacyEnginePerLayer,
+    PrivacyEnginePerLayerNonPoisson,
+)
+
+
+class PrivacyEngineFactory:
+    @classmethod
+    def get(
+        cls,
+        accountant: str = "rdp",  # TODO: str or enum?
```

Comment:
I think Pytorch's style is to accept strings. (eg: loss_reduction); so I vote for str.
My very personal, subjective and highly controversial opinion: str with asserts are better than enums and their implicit checks.

Reply:
`str` is the more pythonic way.
There is a [`Literal[]` type](https://docs.python.org/3/library/typing.html#typing.Literal) to check this. But it is introduced in py3.8. We can't afford use it right now, but we can add them later. Before than, asserts are a good solution. Or probably not just asserts, `if xxx: raise ValueError`.

Reply:
For the reference, [sklearn implementation](https://github.com/scikit-learn/scikit-learn/blob/793e90b344aeee90432b47515410c5666097f0e6/sklearn/linear_model/_logistic.py#L431) of `LogisticRegression(..., solver="lbfgs")`

Reply:
Thanks!

Reply:
TIL about Literal[] type. Die enum die!